src/test/java/test/sanitycheck/SuiteListner.java

```diff
@@ -0,0 +1,28 @@
+package test.sanitycheck;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testng.ISuite;
+import org.testng.ISuiteListener;
+import org.testng.xml.XmlSuite;
+
+public class SuiteListner implements ISuiteListener {
+
+	List<XmlSuite> allSuite = new ArrayList<XmlSuite>();
```

Comment:
> we wont be changing the reference of the array list to any new array list.

Agree, but `final` will warn you if you try ;)
But as I said, it's a personal taste, and I'm ok to not have it in a test class :)